src/execution_plan/ops/op_delete.c

```diff
@@ -201,38 +199,69 @@ static Record DeleteConsume(OpBase *opBase) {
 			break;
 		}
 	}
-
-	return r;
 }
 
-static OpBase *DeleteClone(const ExecutionPlan *plan, const OpBase *opBase) {
-	ASSERT(opBase->type == OPType_DELETE);
-	OpDelete *op = (OpDelete *)opBase;
-	AR_ExpNode **exps;
-	array_clone_with_cb(exps, op->exps, AR_EXP_Clone);
-	return NewDeleteOp(plan, exps);
+// Return mode, emit a populated Record.
+static Record _handoff(OpDelete *op) {
+	return array_pop(op->records);
 }
 
-static OpResult DeleteReset(OpBase *opBase) {
+static Record DeleteConsume(OpBase *opBase) {
 	OpDelete *op = (OpDelete *)opBase;
+	Record r;
 
-	_DeleteEntities(op);
+	// Return mode, all data was consumed.
+	if(op->records) return _handoff(op);
 
-	if(op->deleted_nodes) {
-		array_clear(op->deleted_nodes);
-	}
+	// Consume mode.
+	op->records = array_new(Record, 32);
+	// initialize the records array with NULL, which will terminate execution
+	// upon depletion
+	array_append(op->records, NULL);
 
-	if(op->deleted_edges) {
-		array_clear(op->deleted_edges);
+	GraphContext *gc = QueryCtx_GetGraphCtx();
+	ASSERT(op->op.childCount > 0);
+	// Pull data until child is depleted.
+	OpBase *child = op->op.children[0];
+	while((r = OpBase_Consume(child))) {
+		/* Persist scalars from previous ops before storing the record,
+		* as those ops will be freed before the records are handed off. */
```

Comment:
if you chose to use `/**/` multiline comment, please make sure the `*` are aligned on the same column.